libraries/c_sdk/standard/mqtt/src/iot_mqtt_mutex_wrapper.c

```diff
@@ -0,0 +1,163 @@
+/*
+ * FreeRTOS MQTT SHIM V1.0.0
+ * Copyright(C) 2020 Amazon.com, Inc. or its affiliates.All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this softwareand associated documentation files(the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and /or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions :
+ *
+ * The above copyright noticeand this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http ://www.FreeRTOS.org
+ */
+
+/**
+ * @file iot_mqtt_mutex_wrapper.c
+ * @brief Implements wrapper for mutexes in the shim.
+ */
+/* The config header is always included first. */
+#include "iot_config.h"
+
+/* Standard includes. */
+#include <string.h>
+
+/* Error handling include. */
+#include "private/iot_error.h"
+
+/* MQTT internal includes. */
+#include "private/iot_mqtt_internal.h"
+
+#include "semphr.h"
+
+/*-----------------------------------------------------------*/
+
+bool IotMutex_CreateNonRecursiveMutex( StaticSemaphore_t * pMutex )
+{
+    bool status = false;
+
+    IotMqtt_Assert( pMutex != NULL );
+
+    if( ( xSemaphoreCreateMutexStatic( ( SemaphoreHandle_t ) pMutex ) != NULL ) )
+    {
+        /* Mutex created successfully. */
+        status = true;
+    }
```

Comment:
Log error for failure case?

Reply:
The error is logged where this function is being called.

Reply:
How about moving that error into the function in order to avoid the duplication of similar logs, if the logs are the same?